lib/util/batch.js

```diff
@@ -129,16 +147,28 @@ function joinQueries(queries, timestamp, batchType, consistencyHierarchy, cqlVer
     , consistencyMap = null
     , hasTimestamp = false
     , hasQuery = false
-    , supportsNestedTimestamp = supportsCql31(cqlVersion)
+    , supportsNestedTimestamp = typeof cqlVersion === 'boolean' ? cqlVersion : supportsCql31(cqlVersion)
     , queryConsistencyMap;
   if (!isChildBatch) {
     cql.push('BEGIN ' + getBatchTypeString(batchType) + 'BATCH');
   }
+
+  if (supportsNestedTimestamp) {
```

Comment:
I'd suggest extracting a small method here, to keep the containing method from bloating